### PR TITLE
feat: update project using kg api (#2778)

### PR DESCRIPTION
**Pull request by andre-code**
PR to use kg api to update visibility, keywords and description.

### Contents of this PR:
- Modification of the `updateProject` mutation in the `projectKgApi` to override the `project-kg-metadata` tag when a project is modified. Visibility, keywords and description should be updated in the Project Header immediately after the change.
- Introduction of a component to edit the keywords of a project.
- Introduction of a component to edit project descriptions.
- Adjustment of the visibility component to adapt it to the new endpoint, which now returns errors in the "visibility_level" property.
- Introduction of a new component for online submit input.
- Integration of a storybook for the online submit input component: [Storybook Link](https://renku-ci-ui-2793.dev.renku.ch/storybook/?path=/docs/components-forms-inlinesubmitinput--docs).
- Introduction of a new component to display alerts when updates are submitted to a new branch.

### Cases:
- Successful update
Screenshot 2023-09-20 at 10 46 09

- Error when update due branch protection
Screenshot 2023-09-20 at 10 47 24

- Other errors
Screenshot 2023-09-20 at 10 46 20

/deploy renku-graph=development extra-values=core.sentry.dsn=https://0cc02f3a02d3474d9f7e90915873ab4d@sentry.dev.renku.ch/2,core.sentry.environment=renku-dev,core.sentry.enabled=true #persist

### CODE REVIEW

**Review by lorenzo-cavazzi — COMMENTED** [review]
I started reviewing the PR but I keep getting a 409 error every time I try to update something. Could you check that?

In the meanwhile, I added a comment for a small refactoring that would simplify the RTK query logic by always using local tags on KG APIs (see inline comment)

image

File: client/src/features/project/projectKgApi.ts [hunk 1]
```diff
@@ -144,34 +143,32 @@ export const projectKgApi = createApi({
...
+              dispatch(
+                projectsKgApi.util.invalidateTags([
+                  { type: "project-kg-metadata", id: args.projectId },
+                ])
+              );
+            }
+          })
+          .catch((err) => {
+            return err;
+          });
```
[the remark this answers is not in the document]
Comment (andre-code) [reply client/src/features/project/projectKgApi.ts]: Thanks @lorenzo-cavazzi thats a fair point, I moved the mutation to the projectKgApi

File: client/src/features/project/components/ProjectWarningForMerge.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        The operation was successful, but{" "}
+        <strong>
+          this project requires use of merge requests to make changes.
+        </strong>
+      </p>
+      <p>
+        Create a merge request to bring the changes from{" "}
+        <strong>{remoteBranch}</strong> into <strong>{defaultBranch}</strong> to
+        see the {changeDescription} in your project.
+      </p>
```
Comment (lorenzo-cavazzi) [inline client/src/features/project/components/ProjectWarningForMerge.tsx:35]:
```suggestion
      <p>
        The operation was successful, but this project has protected branches.
      </p>
      <p>
        You need to manually create a merge request from{" "}
        <strong>{remoteBranch}</strong> to <strong>{defaultBranch}</strong> to
        see the new {changeDescription} in your project.
      </p>
```

File: client/src/project/shared/ProjectTag.container.js
Comment (lorenzo-cavazzi) [inline client/src/project/shared/ProjectTag.container.js:1]: This is the last place where the `onProjectTagsChange` was used.
You can remove that from the `Project.js` file, and then `setTags`  from the coordinator.
Please double-check if we can remove something else, I guess also `setVisibility` is not necessary anymore

File: client/src/components/errors/RtkErrorAlert.tsx [hunk 3]
```diff
@@ -23,14 +23,53 @@ import { ErrorAlert } from "../Alert";
...
+  if (
+    "data" in error &&
+    typeof error.data === "object" &&
+    error.data !== null &&
+    "branch" in error.data
+  ) {
+    return (error?.data as UpdateProjectResponse)?.branch;
+  }
+  return undefined;
+}
```
Comment (lorenzo-cavazzi) [inline client/src/components/errors/RtkErrorAlert.tsx:62]: I wonder if we should handle errors specific to just a subset of APIs in smaller components. This function adopts a good approach, but it's used in the more generic `extractRkErrorMessage` component even if it only handles errors specifically from the new KG APIs.

We don't need to change it right now, but it is something to keep in mind.

**Review by lorenzo-cavazzi — CHANGES_REQUESTED** [review]
This works very well now! :raised_hands: 

I added a couple of comments inline, just minor cleanups before merging

File: client/src/project/shared/ProjectTag.container.js
Comment (andre-code) [reply client/src/project/shared/ProjectTag.container.js:1]: We can definitely remove `onProjectTagsChange`, but not `setVisibility` as it is still needed when creating a new project... until the new service to create projects works.

**Pull request merged**